Src/ElasticScale.Client/ShardManagement/ShardMap/ShardMap.cs

```diff
@@ -324,9 +378,64 @@ public override string ToString()
             string connectionString,
             ConnectionOptions options)
         {
-            return OpenConnectionForKeyAsync(key, connectionString, null, options);
+            return OpenConnectionForKeyAsync(key, connectionString, (SqlCredential)null, options);
```

Comment:
Ouch, this is a problem. Dependent code is going to have the same ambiguity issue, so adding the new overload is a breaking change. Sorry that I didn't foresee this.

The simplest solution is to eliminate the override that has `AccessToken` but not `SqlCredential`. In other words, have this set of overrides:

```
OpenConnection(string connectionString) { ... }
OpenConnection(string connectionString, SqlCredential credential) { ... }
// OpenConnection(string connectionString, string accessToken) { ... } // <-- Not this one!! 
OpenConnection(string connectionString, SqlCredential credential, string accessToken) { ... }
```

The more complete solution is to define a new class similar to RangeMappingCreationInfo which assembles all the possible parameters to openconnectionforkey, i.e.:

```
        public sealed class SqlConnectionInfo
        {
            public string ConnectionString { get; set; }
            public SqlCredential Credential { get; set; }
            public string AccessToken { get; set; }
        }
```

```
        public Task<SqlConnection> OpenConnectionForKeyAsync<TKey>(
            TKey key,
            SqlConnectionInfo connectionInfo)
        {
            ...
        }

```

Then we can have the following set of overrides:

```
OpenConnection(string connectionString) { ... }
OpenConnection(string connectionString, SqlCredential credential) { ... }
OpenConnection(SqlConnectionInfo connectionInfo) { ... }
```

Reply:
I prefer the 2nd option because it should result in fewer new overrides now and is also more scalable long-term.

Reply:
done with method 1 as of now. 
Do we really want to go with change 2 as it will be change across all login related files and test cases?

Reply:
You are already changing all login related files. Existing test cases should not be affected since it should not be a breaking change. So it's really not scary.

Reply:
Having SqlCredentials and accessToken together in a public overloaded function seems redundant as they both together can not be not null.  

`OpenConnection(string connectionString, SqlCredential credential, string accessToken) { ... }`